python/tvm/relay/quantize/quantize.py

```diff
@@ -368,11 +283,27 @@ def quantize(graph, params=None, dataset=None):
     ret: Function
         The graph after quantization
     """
-    # TODO(zhiics) Move this to the pass manager.
-    graph = optimize(graph, params)
-
-    graph = annotate(graph)
-    graph = calibrate(graph, dataset)
-    graph = realize(graph)
-    graph = _ir_pass.fold_constant(graph)
-    return graph
+    if params:
+        graph = _bind_params(graph, params)
+
+    mod = _module.Module.from_expr(graph)
+    # Perform "SimplifyInference", "FoldScaleAxis", "FoldConstant", and
+    # "CanonicalizeOps" optimization before quantization.
+    optimize = _transform.Sequential([_transform.SimplifyInference(),
+                                      _transform.FoldConstant(),
+                                      _transform.FoldScaleAxis(),
+                                      _transform.CanonicalizeOps(),
+                                      _transform.FoldConstant()])
+
+    calibrate_pass = _transform.function_pass(calibrate, opt_level=1,
+                                              name="calibrate")
+    _set_conv_counter(0)  # reset counter
+    quantize_seq = _transform.Sequential([_transform.QuantizeAnnotate(),
+                                          calibrate_pass,
+                                          _transform.QuantizeRealize(),
+                                          _transform.FoldConstant()])
+
+    with _transform.PassContext(opt_level=3, required_pass=["calibrate"]):
```

Comment:
what does `required_pass` mean? I thought you do the `calibrate` explicitly in `quantize_seq`

Reply:
`required_pass` means these passes are always enabled because we need to check if a pass is enabled before its execution under a given `opt_level`.